hconfig/mapl3hconfig_get_private.F90

```diff
@@ -0,0 +1,107 @@
+#include "MAPL_ErrLog.h"
+module mapl3hconfig_get_private
+   use :: esmf, only: ESMF_HConfig, ESMF_HConfigIsDefined, MAXSTRLEN => ESMF_MAXSTR
+   use :: esmf, only: ESMF_HConfigAsI4, ESMF_KIND_I4, ESMF_HConfigAsI8, ESMF_KIND_I8
+   use :: esmf, only: ESMF_HConfigAsR4, ESMF_KIND_R4, ESMF_HConfigAsR8, ESMF_KIND_R8
+   use :: esmf, only: ESMF_HConfigAsLogical, ESMF_HConfigAsString
+   use mapl_ErrorHandling
+   use mapl_KeywordEnforcer
+
+   implicit none
+
+   public :: MAXSTRLEN
+   public :: get_value
+
+   interface get_value
+      module procedure :: get_value_scalar
+   end interface get_value
+
+contains
+
+   subroutine get_value_scalar(hconfig, value, found, message, keystring, rc)
+      type(ESMF_HConfig), intent(inout) :: hconfig
+      class(*), intent(inout) :: value
+      logical, intent(out) :: found
+      character(len=:), allocatable, intent(inout) :: message
+      character(len=*), intent(in) :: keystring
+      integer, intent(out) :: rc
+
+      character(len=:), allocatable :: typestring
+      character(len=:), allocatable :: valuestring
+
+      integer :: status
+      integer :: ios
+      character(len=MAXSTRLEN) :: rawstring
+
+      found = ESMF_HConfigIsDefined(hconfig, keystring=keystring, _RC)
+      if(.not. found) then
+         _RETURN(_SUCCESS)
+      end if
+
+      select type(value)
+      type is (integer(kind=ESMF_KIND_I4))
+         typestring = 'I4'
+         value = ESMF_HConfigAsI4(hconfig, keyString=keystring, _RC)
+         write(rawstring, fmt='(I12)', iostat=ios) value
+      type is (integer(kind=ESMF_KIND_I8))
+         typestring = 'I8'
+         value = ESMF_HConfigAsI8(hconfig, keyString=keystring, _RC)
+         write(rawstring, fmt='(I22)', iostat=ios) value
+      type is (real(kind=ESMF_KIND_R4))
+         typestring = 'R4'
+         value = ESMF_HConfigAsR4(hconfig, keyString=keystring, _RC)
+         write(rawstring, fmt='(G17.8)', iostat=ios) value
+      type is (real(kind=ESMF_KIND_R8))
+         typestring = 'R8'
+         value = ESMF_HConfigAsR8(hconfig, keyString=keystring, _RC)
+         write(rawstring, fmt='(G24.16)', iostat=ios) value
+      type is (logical)
+         typestring = 'L'
+         value = ESMF_HConfigAsLogical(hconfig, keyString=keystring, _RC)
+         write(rawstring, fmt='(L1)', iostat=ios) value
```

Comment:
This is the one case you might want to keep the rawstring as there are many ways to spell `true` in yaml.

Reply:
See the comment above.